python/tvm/contrib/hexagon/session.py

```diff
@@ -286,6 +296,11 @@ def _aot_executor_from_factory(
             for target in module.target.values()
             if "hexagon" in target.keys
         )
```

Comment:
Around here, similar to how the hexagon architecture gets pulled out, we should also pull out the target host.  If it is "llvm", we can set `self._requires_cpu_device = True` (or some better-named variable).  Then it can be referred to in the device property

```python
@property
def device(self):
    if hasattr(self, '_device'):
        return self._device

    if self._requires_cpu_device:
         self._device = self._rpc.cpu(0)
    else:
         self._device = self._rpc.hexagon(0)

    return self._device
```

Reply:
this is a much better approach, thanks for the suggestion. I've added it. PTAL